ramp4-pcar4/ramp4-pcar4
# Make horizontal scrollbar visible, Add root element accessor, Fix grid tooltip

### Related Item(s)
#2406 
#2499 
#2375
#2522

### Changes
- Update `ag-grid-vue3` and `ag-grid-community` to enable visibility of horizontal scrollbar 
- Create an attribute `rootEl` for `InstanceAPI`, that represents `$vApp.$root?.$el`
- Make grid tooltip appear exclusively for keyboard users

### QA Testing
Please test against `main` branch https://ramp4-pcar4.github.io/ramp4-pcar4/main/demos/enhanced-samples.html

### Testing

Steps #2406:
1. Open enhanced sample 5 in Firefox
2. Open the grid for the 'Shape' layer
3. Try dragging the horizontal scrollbar at the bottom of the grid, and ensure that it works as intended

Steps #2375:
1. Open enhanced sample 5 in Chrome
2. Open the grid for the 'Shape' layer
3. Use Tab to navigate to the grid panel
4. Enter the panel and tab to the grid
5. Upon arriving at the grid, observe the tooltip, then tab to enter the grid
6. Shift tab out of the grid and observe the tooltip
7. Click on the area to the left of the horizontal scrollbar (this brings focus to the grids container)
![image](https://github.com/user-attachments/assets/9e47f731-6933-4140-b09b-373002d9594a)
8. Observe that the tooltip doesn't appear anymore
9. Go into mobile mode and scroll throughout the grid
10. Observe that the tooltip doesn't appear

Steps #2499:
- Since only the export panel was using `iApi.$vApp.$root?.$el` to access the root element (and was thus changed to use `rootEl`), test to ensure that the export panel is working correctly


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/2515)
